docs/en/development/CI.md

```diff
@@ -0,0 +1,209 @@
+## CI checks
+
+When you submit a pull request, some automated checks are ran for your code by
+the ClickHouse [continuous integration (CI) system](tests.md#test-automation).
+This happens after a repository maintainer (someone from ClickHouse team) has
+screened your code and added the `can be tested` label to your pull request.
+The results of the checks are listed on the GitHub pull request page as
+described in the [GitHub checks
+documentation](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-status-checks).
+If a check is failing, you might be required to fix it. This page gives an
+overview of checks you may encounter, and what you can do to fix them.
+
+If it looks like the check failure is not related to your changes, it may be
+some transient failure or an infrastructure problem. Push an empty commit to
+the pull request to restart the CI checks:
+```
+git reset
+git commit --allow-empty
+git push
+```
+
+If you are not sure what to do, ask a maintainer for help.
+
+
+### Merge with master
+
+Verifies that the PR can be merged to master. If not, it will fail with the
+message 'Cannot fetch mergecommit'. To fix this check, resolve the conflict as
+described in the [GitHub
+documentation](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/resolving-a-merge-conflict-on-github),
+or merge the `master` branch to your pull request branch using git.
```

Comment:
better rebase

Reply:
I don't care whether you merge or rebase. Both are Ok.
Rebase also have some downsides. If you rebase, check results for previous commits will be lost.